index.js

```diff
@@ -1,32 +1,50 @@
 /* eslint-env node */
 'use strict';
 
-var path = require('path');
-var filterInitializers = require('fastboot-filter-initializers');
-var VersionChecker = require('ember-cli-version-checker');
-var mergeTrees = require('broccoli-merge-trees');
+const path = require('path');
+const filterInitializers = require('fastboot-filter-initializers');
+const VersionChecker = require('ember-cli-version-checker');
+const mergeTrees = require('broccoli-merge-trees');
 
 module.exports = {
   name: 'ember-cli-head',
 
-  treeForApp: function(defaultTree) {
-    var checker = new VersionChecker(this);
-    var emberVersion = checker.for('ember-source', 'npm');
+  treeForApp(defaultTree) {
+    if (!process.env.FASTBOOT_NEW_BUILD) {
+      let trees = [defaultTree];
 
-    if (!emberVersion.version) {
-      emberVersion = checker.for('ember', 'bower');
+      if (!this.hasGlimmer2()) {
+        trees.push(this.treeGenerator(path.resolve(this.root, 'app-lt-2-10')));
+        trees.push(this.treeGenerator(path.resolve(this.root, 'fastboot-lt-2-10')));
+      }
+
+      let tree = mergeTrees(trees, { overwrite: true });
+
+      return filterInitializers(tree);
+    } else {
+      return defaultTree;
     }
+  },
 
-    var trees = [defaultTree];
+  addToFastBootTree() {
+    // this hook will be invoked in post FastBoot 1.0
+    let fastbootBuilder = require('ember-cli-fastboot/lib/build-utilities/fastboot-builder');
```

Comment:
You don't need this any longer. `treeForFastBoot` now takes a `treeGenerator`. See an example [here](https://github.com/ember-fastboot/ember-cli-fastboot/blob/master/index.js#L109)